src/Command/MergeAssetsAndVariationsFilesCommand.php

```diff
@@ -0,0 +1,252 @@
+                ]
+            );
+        } catch (IOException|UnsupportedTypeException|ReaderNotOpenedException $e) {
+            $this->io->error($e->getMessage());
+
+            exit;
+        }
+
+        $this->io->text('');
+
+        $this->io->text('Now merging files and create new Assets...');
+        $this->retrieveChannelsAndLocales();
+        $this->mergeFiles($output, $targetFilePath);
+        $this->io->success(sprintf('%s assets created in "%s"', $this->assetsReader->count(), $targetFilePath));
+    }
+
+    private function mergeFiles(OutputInterface $output, string $targetFilePath)
+    {
+        $progressBar = new ProgressBar($output, $this->assetsReader->count());
+        $progressBar->start();
+
+        /** @var Writer $writer */
+        $writer = WriterFactory::create(Type::CSV);
+        $writer->setFieldDelimiter(self::CSV_FIELD_DELIMITER);
+        $writer->setFieldEnclosure(self::CSV_FIELD_ENCLOSURE);
+        $writer->openToFile($targetFilePath);
+        $writer->addRow($this->getAssetManagerFileHeaders());
+
+        foreach ($this->assetsReader as $assetLineNumber => $row) {
+            if ($assetLineNumber === 1) {
+                continue;
+            }
+
+            if (!$this->isHeaderValid($this->assetsReader, $row)) {
+                continue;
+            }
+
+            $assetLine = array_combine($this->assetsReader->getHeaders(), $row);
+
+            $variationsForThisAsset = [];
+            foreach ($this->variationsReader as $variationLineNumber => $variationRow) {
+                if ($variationLineNumber === 1) {
+                    continue;
+                }
+
+                if (!$this->isHeaderValid($this->variationsReader, $variationRow)) {
+                    continue;
+                }
+
+                $variationLine = array_combine($this->variationsReader->getHeaders(), $variationRow);
+
+                if ($variationLine['asset'] === $assetLine['code']) {
+                    $variationsForThisAsset[] = $variationLine;
+                }
+            }
+
+            $newAsset = $this->mergeAssetAndVariations($assetLine, $variationsForThisAsset);
+
+            $writer->addRow($newAsset);
+            $progressBar->advance();
+        }
+
+        $writer->close();
+        $progressBar->finish();
+    }
+
+    private function isHeaderValid(CsvReader $reader, $row)
+    {
+        return count($reader->getHeaders()) === count($row);
+    }
+
+    /**
+     * The new file Headers are a merge between the Asset CSV file headers and the generated value keys
+     * computed from the Variation CSV file headers (attribute-channel-locale)
+     */
+    private function getAssetManagerFileHeaders(): array
+    {
+        $assetHeaders = $this->assetsReader->getHeaders();
+        $valuesHeaders = [];
+        foreach ($this->channels as $channel) {
+            $valuesHeaders[] = sprintf('reference_file-%s', $channel);
```

Comment:
I would suggest to use a constant for `reference_file` as this may be configurable in the future :)